# Task 2 - Warm Up Vaccine Data

In the tasks below, I use the vaccine data set and the sheet Vaccinated municipality and age.
I will use Plotly express to draw the diagrams below for the tasks that require diagrams and use Pandas
to answer questions about the data set.

In [13]:
import pandas as pd

In [14]:
data = pd.read_excel("Data/Folkhalsomyndigheten_Covid19_Vaccine.xlsx", sheet_name="Vaccinerade kommun och ålder")
df = pd.DataFrame(data)
df.head(3)

Län        Län_namn  Kommun     Kommun_namn  Ålder  Befolkning  \
0    1  Stockholms län     114  Upplands Väsby  12-15        2422   
1    1  Stockholms län     114  Upplands Väsby  16-17        1203   
2    1  Stockholms län     114  Upplands Väsby  18-29        6692   

   Antal minst 1 dos  Antal minst 2 doser  Antal 3 doser  Antal 4 doser  \
0               1206                 1046            NaN            NaN   
1                839                  755            NaN            NaN   
2               4887                 4469         1959.0            NaN   

   Andel minst 1 dos  Andel minst 2 doser  Andel 3 doser  Andel 4 doser  
0           0.497936             0.431874            NaN            NaN  
1           0.697423             0.627598            NaN            NaN  
2           0.730275             0.667812       0.292738            NaN

### **a)** How many counties are represented in the data set?

In [15]:
print(f"The number of counties in the dataset is: {len(df['Län_namn'].value_counts())}. It is the following counties:\n{df['Län_namn'].value_counts()}")

The number of counties in the dataset is: 21. It is the following counties:
Län_namn
Västra Götalands län    490
Skåne län               330
Stockholms län          260
Värmlands län           160
Västerbottens län       150
Dalarnas län            150
Norrbottens län         140
Jönköpings län          130
Östergötlands län       130
Kalmar län              120
Örebro län              120
Västmanlands län        100
Gävleborgs län          100
Södermanlands län        90
Uppsala län              80
Kronobergs län           80
Jämtlands län            80
Västernorrlands län      70
Hallands län             60
Blekinge län             50
Gotlands län             10
Name: count, dtype: int64


In the above code we can see that all 21 of Swedens counties are represented in the data set.

Source: [SKR](https://skr.se/skr/tjanster/kommunerochregioner/faktakommunerochregioner.1022.html)

### **b)** How many municipalities are represented in the data set?

In [16]:
print(f"The number of municipalities in the dataset is: {len(df['Kommun_namn'].value_counts())}.")

The number of municipalities in the dataset is: 290.


In the above code we can see that all 290 of Swedens municipalities are represented in the dataset.

Source: [SKR](https://skr.se/skr/tjanster/kommunerochregioner/faktakommunerochregioner.1022.html)

### **c)** How large is the population represented in the data set?


In [18]:
print(f"The total sum of the population represented in the data set is: {df['Befolkning'].sum()}.")

The total sum of the population represented in the data set is: 9092790.


According to [SCB](https://www.scb.se/hitta-statistik/statistik-efter-amne/befolkning/befolkningens-sammansattning/befolkningsstatistik/pong/tabell-och-diagram/folkmangd-och-befolkningsforandringar---manad-kvartal-och-halvar/befolkningsstatistik-2023/) (Statistiska Centralbyrån) the total population of Sweden in oktober 2022 where 
10,519,592 but in our data set only 9,092,790 is represented. In the code below we can see that there is no NaN/Null values in the 'Befolkning column' so the reason in not because of values missing.

In [21]:
df.isnull().sum()

Län                       0
Län_namn                  0
Kommun                    0
Kommun_namn               0
Ålder                     0
Befolkning                0
Antal minst 1 dos         0
Antal minst 2 doser       0
Antal 3 doser           580
Antal 4 doser          2030
Andel minst 1 dos         0
Andel minst 2 doser       0
Andel 3 doser           580
Andel 4 doser          2030
dtype: int64

### **d)** Calculate how many children under the age of 18 there are in Sweden based on this data set. 

First I need to find the unique values in the data set using the .unique() method on the column 'Ålder' (age).

In [22]:
df['Ålder'].unique()

array(['12-15', '16-17', '18-29', '30-39', '40-49', '50-59', '60-69',
       '70-79', '80-89', '90 eller äldre'], dtype=object)

The above code indicates that the age groups for children includes children aged 12 to 17 years. Down below we will include the two groups below 18 in our .query() method and makning a new dataframe called 'children_non_legal_age'. Then we will aggregate the total sum of pupulation from our new dataframe.

In [39]:
children_non_legal_age = df.query("Ålder == ['12-15', '16-17']")
number_of_children_12_to_17 = children_non_legal_age['Befolkning'].sum()
number_of_children_12_to_17

745370

If we assume that the births in Sweden after the year 2010 to 2022 were approximately the same as for the cohorts of children who were 12-17 in 2022. Would that mean that the **children represented in the data set** (six cohorts) **times 2** (twelve cohorts [aged 0-11] ) would be **representative** of the children who were **0-11 years old** at the time of completion of the data collection. 

If we then add the children missing (0-11) to the total sum of the population in our dataset (9,092 790) we actually get a value more similar to  data from SCB refered to in question c) (10,519,592).

In [43]:
number_of_children_0_to_11 = number_of_children_12_to_17 * 2
total_population = number_of_children_0_to_11 + df['Befolkning'].sum()
total_population

10583530

In [45]:
children_under_18 = number_of_children_0_to_11 + number_of_children_12_to_17
children_under_18

2236110

As seen above the total number of children younger than 18 years would be 2236110 and this divided by the total population 10583530 would give us the result that approximately efter fifth person in Sweden were below 18 years old in the autum of 2022.

In [56]:
print(f"The children below 18 were {(children_under_18/total_population) * 100:.2f}% in the autum of 2022 according to our data set.")

The children below 18 were 21.13% in the autum of 2022 according to our data set.
